### PR TITLE
Configure and enable core uploader

### PR DESCRIPTION
**Pull request by renukamanavalan**
If secret key for core file uploader is available, configure it, enable & start the service.
Also program https-proxy if available for the core file uploader service.

Tested in private internal branch.

### CODE REVIEW

**Comment by lguohan** [comment]
https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-image-pr/3108/console

this is the error message for previous pr.

what is corefile_uploader['env'] is not defined?

**Comment by renukamanavalan** [comment]
The error was not because 'env' is not defined, but overloaded quotes. It is a syntax error.
line: '\1https_proxy": "{{ corefile_uploader['env']['https_proxy'] }}"'

The entire line is wrapped in single quote, and I used single quote inside too, around env, which totally compiles out as 
'\1https_proxy": "{{ corefile_uploader['   +  env  + ']['https_proxy'] }}"'
Here env is expected as variable/fact which is not the case.

In new code, I read it separately and use the read value to fill, so no too many quotes issue.
The code explicitly check for "when: corefile_uploader['env']['https_proxy'] is defined"

**Comment by renukamanavalan** [comment]
In fact, it compiled out as 
'\1https_proxy": "{{ corefile_uploader[' + env + '][' + https_proxy + '] }}"'

**Comment by lguohan** [comment]
can you provide sample var file in group_vars/all/corefile_uploader.yml

just commented so that if external user want to use, they know how to provide those data?

**Comment by renukamanavalan** [comment]
Sample:  group_vars/all/secrets.json
...
    "corefile_uploader": {
        "azure_sonic_core_storage": {
                "account_key": "..."
           },
        "env": {
            "https_proxy": "..."
        }
...

**Comment by lguohan** [comment]
i mean as part of the pr.

File: ansible/group_vars/all/corefile_uploader.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Configure core file storage secret key and https-proxy as required
+#
+#corefile_uploader:
+#        azure_sonic_core_storage:
+#                account_key: "Your Secret"
+#
```
[the remark this answers is not in the document]
Comment (renukamanavalan) [reply ansible/group_vars/all/corefile_uploader.yml:6]: line 6 ? 
I will.

File: ansible/config_sonic_basedon_testbed.yml [hunk 2]
```diff
@@ -144,6 +144,49 @@
...
+
+      - name: read account key
+        set_fact:
+            core_key: "{{ corefile_uploader['azure_sonic_core_storage']['account_key'] }}"
+        when: corefile_uploader['azure_sonic_core_storage']['account_key'] is defined
+
+      - name: read https proxy
+        set_fact:
+            core_proxy: "{{ corefile_uploader['env']['https_proxy'] }}"
+        when: corefile_uploader['env']['https_proxy'] is defined
```
[the remark this answers is not in the document]
Comment (lguohan) [reply ansible/config_sonic_basedon_testbed.yml:160]: so you are saying if corefile_uploader does not exist in the group var files, the script here will proceed, will not fail, right?

[the remark this answers is not in the document]
Comment (renukamanavalan) [reply ansible/config_sonic_basedon_testbed.yml:160]: yes, unless your switch requires proxy.

File: ansible/group_vars/all/corefile_uploader.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (renukamanavalan) [reply ansible/group_vars/all/corefile_uploader.yml]: done

**Pull request merged**